core/flyout_horizontal.js

```diff
@@ -236,44 +238,103 @@ Blockly.HorizontalFlyout.prototype.wheel_ = function(e) {
  * @protected
  */
 Blockly.HorizontalFlyout.prototype.layout_ = function(contents, gaps) {
-  this.workspace_.scale = this.targetWorkspace.scale;
+  Blockly.Flyout.prototype.layout_.call(this, contents, gaps);
+  this.layoutContent_();
+};
+
+/**
+ * Respositions the blocks, buttons and labels in the flyout based on their
+ *     current dimensions and order.
+ * @protected
+ */
+Blockly.HorizontalFlyout.prototype.layoutContent_ = function() {
+  this.workspace_.setResizesEnabled(false);
+  this.reflow();
   var margin = this.MARGIN;
   var cursorX = margin + this.tabWidth_;
   var cursorY = margin;
+
+  var contents = this.contents_.slice();
+  var gaps = this.gaps_.slice();
+  // For RTL in a X axis layout, if the length of the blocks is less than the
+  // width of the flyout, the blocks need to be offset so that they appear
+  // right-aligned. To do so, we calculate the width of the blocks (and offset,
+  // if any) before adjusting the blocks' positions.
   if (this.RTL) {
-    contents = contents.reverse();
+    contents.reverse();
+    gaps.reverse();
+
+    var contentWidth = cursorX;
+    for (var i = 0, item; (item = contents[i]); i++) {
+      if (item instanceof Blockly.BlockSvg) {
+        contentWidth += item.getHeightWidth().width;
+      } else if (item instanceof Blockly.FlyoutButton) {
+        contentWidth += item.width;
+      }
+      contentWidth += gaps[i];
+    }
+    var delta = this.getWidth() - contentWidth;
+    if (delta > 0) {
+      cursorX += delta;
```

Comment:
Does this function fix a different bug with the flyout? Or is this still trying to fix the scooching blocks over part?